paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/linkinline/LinkInlineSignupConfirmationDefinition.kt

```diff
@@ -0,0 +1,191 @@
+
+                linkInlineSignupConfirmationOption.toNewOption()
+            }
+            AccountStatus.SignedOut,
+            AccountStatus.Error -> {
+                linkConfigurationCoordinator.signInWithUserInput(configuration, userInput).fold(
+                    onSuccess = {
+                        // If successful, the account was fetched or created, so try again
+                        createPaymentMethodConfirmationOption(linkInlineSignupConfirmationOption)
+                    },
+                    onFailure = {
+                        linkInlineSignupConfirmationOption.toNewOption()
+                    }
+                )
+            }
+        }
+    }
+
+    private suspend fun createOptionAfterAttachingToLink(
+        linkInlineSignupConfirmationOption: LinkInlineSignupConfirmationOption,
+    ): PaymentMethodConfirmationOption {
+        val createParams = linkInlineSignupConfirmationOption.createParams
+        val saveOption = linkInlineSignupConfirmationOption.saveOption
+
+        val linkPaymentDetails = linkConfigurationCoordinator.attachNewCardToAccount(
+            linkInlineSignupConfirmationOption.linkConfiguration,
+            createParams,
+        ).getOrNull()
+
+        return when (linkPaymentDetails) {
+            is LinkPaymentDetails.New -> {
+                linkStore.markLinkAsUsed()
+
+                PaymentMethodConfirmationOption.New(
+                    createParams = linkPaymentDetails.paymentMethodCreateParams,
+                    optionsParams = PaymentMethodOptionsParams.Card(
+                        setupFutureUsage = saveOption.setupFutureUsage
+                    ),
+                    shouldSave = saveOption.shouldSave(),
+                )
+            }
+            is LinkPaymentDetails.Saved -> {
+                linkStore.markLinkAsUsed()
+
+                val last4 = linkPaymentDetails.paymentDetails.last4
+
+                PaymentMethodConfirmationOption.Saved(
+                    paymentMethod = PaymentMethod.Builder()
+                        .setId(linkPaymentDetails.paymentDetails.id)
+                        .setCode(createParams.typeCode)
+                        .setCard(
+                            PaymentMethod.Card(
+                                last4 = last4,
+                                wallet = Wallet.LinkWallet(last4),
+                            )
+                        )
+                        .setType(PaymentMethod.Type.Card)
+                        .build(),
+                    optionsParams = PaymentMethodOptionsParams.Card(
+                        setupFutureUsage = ConfirmPaymentIntentParams.SetupFutureUsage.OffSession?.takeIf {
+                            saveOption.shouldSave()
+                        } ?: ConfirmPaymentIntentParams.SetupFutureUsage.Blank
+                    ),
+                )
+            }
+            null -> linkInlineSignupConfirmationOption.toNewOption()
```

Comment:
How come only this one is broken out into a method? Should we break them all out?